tests/go/kgo-verifier/pkg/worker/worker.go

```diff
@@ -244,6 +244,8 @@ func (wc *WorkerConfig) MakeKgoOpts() []kgo.Opt {
 		kgo.ProducerBatchMaxBytes(int32(wc.BatchMaxbytes)),
 		kgo.MaxBufferedRecords(int(wc.MaxBufferedRecords)),
 		kgo.RequiredAcks(kgo.AllISRAcks()),
+
+		kgo.AlwaysRetryEOF(), // workaround for CORE-14849
```

Comment:
There are many `NewClient` calls in these binaries, but luckily most defer here to `MakeKgoOpts`. I only searched for `NewClient` calls: hopefully that is sufficient.